### PR TITLE
feat(Tooltip): introduced widthMode

### PR DESCRIPTION
**Pull request by MattSchiller**
### Overview

In working on adding a tooltip for the profile cards for contributors in Docs in [this PR](https://github.com/codecademy-engineering/portal-app/pull/683), we found that there were some things that needed to be added to the tooltip:
- ~The ability to hide the tooltip even when hovered/focused at certain breakpoints~
- ~The ability to turn off the labelled by functionality since this caused a11y issues for different tooltip contents~
- The ability to override the actual displayed width

I simply wasn't able to get the width of the tooltip to get wider, even when I gave my inner contents box greater width wrapped in a box:

Screen Shot 2021-09-21 at 4 29 21 PM

I realize this code is a bit suboptimal, but the tooltip seems to not be capable of doing the (reasonable?) things that we're asking of it.

I'd like to circle back and update the tooltip and popover components [ticket here](https://codecademy.atlassian.net/browse/REACH-1339), but I simply don't have the bandwidth right now.

If there's a different combination of CSS/markup at the callsite that's available, please, please let me know what about the Gamut/CSS render I'm missing and I'll be happy to forego this intermediary props hack.

Adds optional styling escape hatch (that might not be needed) & ability to turn off the aria-labelledby functionality.

### PR Checklist

- [ ] Related to designs:
- [X] Related to JIRA ticket: REACH-1268
- [X] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.bb0a6c.0<br/>@codecademy/gamut-labs@23.0.1-alpha.bb0a6c.0<br/>@codecademy/gamut@37.6.3-alpha.bb0a6c.0<br/>@codecademy/monaco@0.4.11-alpha.bb0a6c.0<br/>@codecademy/styleguide@46.0.1-alpha.bb0a6c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.eb73a3.0<br/>@codecademy/gamut-labs@23.0.1-alpha.eb73a3.0<br/>@codecademy/gamut@37.6.3-alpha.eb73a3.0<br/>@codecademy/monaco@0.4.11-alpha.eb73a3.0<br/>@codecademy/styleguide@46.0.1-alpha.eb73a3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614a26113fdc8800cc0ebe75--gamut-preview.netlify.app>https://614a26113fdc8800cc0ebe75--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614a26113fdc8800cc0ebe75>Deploy Logs</a></p>

File: packages/gamut/src/ToolTip/index.tsx [hunk 1]
```diff
@@ -238,8 +274,14 @@ export const ToolTip: React.FC<ToolTipProps> = ({
         role="tooltip"
         mode={mode}
         aria-live="polite"
+        breakpointVisibility={breakpointVisibility}
       >
-        <ToolTipBody alignment={alignment} mode={mode}>
+        <ToolTipBody
+          alignment={alignment}
+          mode={mode}
+          minWidth="4rem"
```
Comment (MattSchiller) [inline packages/gamut/src/ToolTip/index.tsx:282]: moved `minWidth` down because it wasn't being very friendly. Very much open to functional alternatives.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.f8d2e5.0<br/>@codecademy/gamut-labs@23.0.1-alpha.f8d2e5.0<br/>@codecademy/gamut@37.6.3-alpha.f8d2e5.0<br/>@codecademy/monaco@0.4.11-alpha.f8d2e5.0<br/>@codecademy/styleguide@46.0.1-alpha.f8d2e5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614a4803488d8d214188f7a2--gamut-preview.netlify.app>https://614a4803488d8d214188f7a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614a4803488d8d214188f7a2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.35d24a.0<br/>@codecademy/gamut-labs@23.0.1-alpha.35d24a.0<br/>@codecademy/gamut@37.6.3-alpha.35d24a.0<br/>@codecademy/monaco@0.4.11-alpha.35d24a.0<br/>@codecademy/styleguide@46.0.1-alpha.35d24a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614a4b3bd6d65a329ae78940--gamut-preview.netlify.app>https://614a4b3bd6d65a329ae78940--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614a4b3bd6d65a329ae78940>Deploy Logs</a></p>

File: packages/gamut/src/ToolTip/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/ToolTip/index.tsx:284]: In what way?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.807b0a.0<br/>@codecademy/gamut-labs@23.0.1-alpha.807b0a.0<br/>@codecademy/gamut@37.6.3-alpha.807b0a.0<br/>@codecademy/monaco@0.4.11-alpha.807b0a.0<br/>@codecademy/styleguide@46.0.1-alpha.807b0a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61525888b378f61b8c2d5de9--gamut-preview.netlify.app>https://61525888b378f61b8c2d5de9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61525888b378f61b8c2d5de9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.d1d545.0<br/>@codecademy/gamut-labs@23.0.1-alpha.d1d545.0<br/>@codecademy/gamut@37.6.3-alpha.d1d545.0<br/>@codecademy/monaco@0.4.11-alpha.d1d545.0<br/>@codecademy/styleguide@46.0.1-alpha.d1d545.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.85f983.0<br/>@codecademy/gamut-labs@23.0.1-alpha.85f983.0<br/>@codecademy/gamut@37.6.3-alpha.85f983.0<br/>@codecademy/monaco@0.4.11-alpha.85f983.0<br/>@codecademy/styleguide@46.0.1-alpha.85f983.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6153370bba4b425a0cbfcb1a--gamut-preview.netlify.app>https://6153370bba4b425a0cbfcb1a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6153370bba4b425a0cbfcb1a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.bc1ce8.0<br/>@codecademy/gamut-labs@23.0.1-alpha.bc1ce8.0<br/>@codecademy/gamut@37.6.3-alpha.bc1ce8.0<br/>@codecademy/monaco@0.4.11-alpha.bc1ce8.0<br/>@codecademy/styleguide@46.0.1-alpha.bc1ce8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61533d0d64a12e6e7c84239c--gamut-preview.netlify.app>https://61533d0d64a12e6e7c84239c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61533d0d64a12e6e7c84239c>Deploy Logs</a></p>

File: packages/gamut/src/ToolTip/index.tsx (hunk 1 above)
Comment (MattSchiller) [reply packages/gamut/src/ToolTip/index.tsx:269]: Due to the priority of styling, any external overrides being passed into the component were being overridden themselves by the styled component rules. By pulling it out, `toolTipStyles` could override the `minWidth` value.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.924b4d.0<br/>@codecademy/gamut-labs@23.0.1-alpha.924b4d.0<br/>@codecademy/gamut@37.6.3-alpha.924b4d.0<br/>@codecademy/monaco@0.4.11-alpha.924b4d.0<br/>@codecademy/styleguide@46.0.1-alpha.924b4d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://615387fc1b8f48504319b113--gamut-preview.netlify.app>https://615387fc1b8f48504319b113--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/615387fc1b8f48504319b113>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.49e758.0<br/>@codecademy/gamut-labs@23.0.1-alpha.49e758.0<br/>@codecademy/gamut@37.6.3-alpha.49e758.0<br/>@codecademy/monaco@0.4.11-alpha.49e758.0<br/>@codecademy/styleguide@46.0.1-alpha.49e758.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://615489a7ee75b02553f6444f--gamut-preview.netlify.app>https://615489a7ee75b02553f6444f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/615489a7ee75b02553f6444f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.94-alpha.81f4b3.0<br/>@codecademy/gamut-labs@23.3.1-alpha.81f4b3.0<br/>@codecademy/gamut@37.6.4-alpha.81f4b3.0<br/>@codecademy/monaco@0.4.12-alpha.81f4b3.0<br/>@codecademy/styleguide@46.4.1-alpha.81f4b3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61549a16fd5c2c00e688132a--gamut-preview.netlify.app>https://61549a16fd5c2c00e688132a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61549a16fd5c2c00e688132a>Deploy Logs</a></p>

**Pull request merged**